packages/tracing/test/span.test.ts

```diff
@@ -473,6 +472,34 @@ describe('Span', () => {
       });
       expect(baggage && getThirdPartyBaggage(baggage)).toStrictEqual('');
     });
+
+    describe('Including transaction name in DSC', () => {
+      test.each([
+        ['is not included if transaction source is not set', undefined, false],
+        ['is not included if transaction source is url', 'url', false],
+        ['is not included if transaction source is unknown', 'unknown', false],
+        ['is included if transaction source is paremeterized route/url', 'route', true],
+        ['is included if transaction source is a custom name', 'custom', true],
+      ])('%s', (_: string, source, shouldBeIncluded) => {
+        const transaction = new Transaction(
+          {
+            name: 'tx',
+            metadata: {
+              ...(source && { source: source as TransactionSource }),
+            },
+          },
+          hub,
+        );
+
+        const dsc = getSentryBaggageItems(transaction.getBaggage());
+
+        if (shouldBeIncluded) {
+          expect(dsc.transaction).toEqual('tx');
+        } else {
+          expect(dsc.transaction).toBeUndefined();
+        }
```

Comment:
Conditional `expect`s are a bit of an antipattern because they make tests more complicated than they need to be. I think we can get away with it here though.

Reply:
You're right, this is fishy. lemme split the two scenarios up, then we should be good